src/PointwiseFunctions/AnalyticSolutions/ScalarAdvection/Sinusoid.cpp

```diff
@@ -0,0 +1,48 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/ScalarAdvection/Sinusoid.hpp"
+
+#include <cmath>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace ScalarAdvection::Solutions {
+
+template <typename DataType>
+tuples::TaggedTuple<ScalarAdvection::Tags::U> Sinusoid::variables(
+    const tnsr::I<DataType, 1>& x, double t,
+    tmpl::list<ScalarAdvection::Tags::U> /*meta*/) const noexcept {
+  auto u = make_with_value<Scalar<DataType>>(get<0>(x), 0.0);
+  get(u) = sin(M_PI * (get<0>(x) - t));
+  return u;
+}
+
+void Sinusoid::pup(PUP::er& /*p*/) noexcept {}
+
+bool operator==(const Sinusoid& /*lhs*/, const Sinusoid& /*rhs*/) noexcept {
+  return true;
+}
+
+bool operator!=(const Sinusoid& lhs, const Sinusoid& rhs) noexcept {
+  return not(lhs == rhs);
+}
+
+}  // namespace ScalarAdvection::Solutions
+
+#define DTYPE(data) BOOST_PP_TUPLE_ELEM(0, data)
+
+#define INSTANTIATE(_, data)                             \
+  template tuples::TaggedTuple<ScalarAdvection::Tags::U> \
+  ScalarAdvection::Solutions::Sinusoid::variables(       \
+      const tnsr::I<DTYPE(data), 1>& x, double t,        \
+      tmpl::list<ScalarAdvection::Tags::U> /*meta*/) const noexcept;
+
+GENERATE_INSTANTIATIONS(INSTANTIATE, (DataVector))
```

Comment:
@nilsdeppe I see other analytic solution templates are instantiated with both `double` and `DataVector`, while in this case `double` would not work since the scalar U field defined in `Evolution/Systems/ScalarAdvection/Tags` has DataVector as its fixed dtype. Do you think I need to make corresponding changes to incorporate `double`?

Reply:
No, I don't think we'll ever need to use this solution with a `double`. If so, probably the best approach would be to template the tag on the data type, but no need to do that now.

Reply:
Okay!